### PR TITLE
Delete Payment Changes

### PR DESCRIPTION
**Pull request by pksumesh**
*Issue #:*
https://github.com/bcgov/entity/issues/1532
https://github.com/bcgov/entity/issues/1862

*Description of changes:*
- Changing the delete endpoint to return 202 and mark as DELETE_ACCEPTED
- Changes to Job to delete the invoice in PayBC

*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [x] No lint errors
- [x] No test case failures and proper coverage for all modules/classes
- [x] Updated the deployment configs for new environment variable(s)
- [ ] Updated the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

**Comment by pksumesh** [comment]
```
(venv) Sumeshs-MacBook-Pro:pay-api aot$ make pylint
pylint --rcfile=setup.cfg \
	 --load-plugins=pylint_flask \
	 --disable=C0301,W0511 \
	 src/pay_api

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

```

**Comment by pksumesh** [comment]
```
---------- coverage: platform darwin, python 3.7.1-final-0 -----------
Name                                            Stmts   Miss Branch BrPart  Cover
---------------------------------------------------------------------------------
src/pay_api/__init__.py                            41      0      2      1    98%
src/pay_api/exceptions/__init__.py                 13      0      0      0   100%
src/pay_api/factory/__init__.py                     1      0      0      0   100%
src/pay_api/factory/payment_system_factory.py      40      1     16      1    96%
src/pay_api/jwt_wrapper.py                         10      0      2      1    92%
src/pay_api/models/__init__.py                     19      0      0      0   100%
src/pay_api/models/audit.py                        12      0      0      0   100%
src/pay_api/models/base_model.py                   15      0      0      0   100%
src/pay_api/models/base_schema.py                  13      0     10      0   100%
src/pay_api/models/code_table.py                    5      0      0      0   100%
src/pay_api/models/corp_type.py                    13      0      0      0   100%
src/pay_api/models/db.py                            5      0      0      0   100%
src/pay_api/models/fee_code.py                     13      0      0      0   100%
src/pay_api/models/fee_schedule.py                 35      0      4      0   100%
src/pay_api/models/filing_type.py                  13      0      0      0   100%
src/pay_api/models/invoice.py                      54      0      8      0   100%
src/pay_api/models/invoice_reference.py            20      0      0      0   100%
src/pay_api/models/payment.py                      36      1      0      0    97%
src/pay_api/models/payment_account.py              26      0      2      0   100%
src/pay_api/models/payment_line_item.py            30      0      0      0   100%
src/pay_api/models/payment_method.py               13      0      0      0   100%
src/pay_api/models/payment_system.py               13      0      0      0   100%
src/pay_api/models/payment_transaction.py          38      0      0      0   100%
src/pay_api/models/receipt.py                      23      0      2      0   100%
src/pay_api/models/status_code.py                  13      0      0      0   100%
src/pay_api/resources/__init__.py                  27      0      0      0   100%
src/pay_api/resources/fee.py                       28      0      2      0   100%
src/pay_api/resources/invoice.py                   31      0      4      0   100%
src/pay_api/resources/invoice_receipt.py           36      0      4      0   100%
src/pay_api/resources/meta.py                      10      0      2      0   100%
src/pay_api/resources/ops.py                       18      0      4      0   100%
src/pay_api/resources/payment.py                   80      2     14      1    97%
src/pay_api/resources/status.py                    20      0      2      0   100%
src/pay_api/resources/transaction.py               64      2      6      0    97%
src/pay_api/schemas/__init__.py                     1      0      0      0   100%
src/pay_api/services/__init__.py                    6      0      0      0   100%
src/pay_api/services/auth.py                       18      0      8      1    96%
src/pay_api/services/base_payment_system.py        23      0      0      0   100%
src/pay_api/services/bcol_service.py               27      0      0      0   100%
src/pay_api/services/fee_schedule.py              106      0     10      0   100%
src/pay_api/services/internal_pay_service.py       28      0      0      0   100%
src/pay_api/services/invoice.py                   218      0     16      0   100%
src/pay_api/services/invoice_reference.py          84      2      4      0    98%
src/pay_api/services/oauth_service.py              64      2     16      2    95%
src/pay_api/services/paybc_service.py             127      0     18      1    99%
src/pay_api/services/payment.py                   125      0      6      0   100%
src/pay_api/services/payment_account.py           110      0      6      0   100%
src/pay_api/services/payment_line_item.py         139      0      2      0   100%
src/pay_api/services/payment_service.py           200      0     46      1    99%
src/pay_api/services/payment_transaction.py       241      0     44      0   100%
src/pay_api/services/queue_publisher.py            31      0      2      0   100%
src/pay_api/services/receipt.py                   105      1     10      1    98%
src/pay_api/services/status_service.py             63      0     26      0   100%
src/pay_api/utils/__init__.py                       1      0      0      0   100%
src/pay_api/utils/auth.py                           3      0      0      0   100%
src/pay_api/utils/constants.py                     11      0      0      0   100%
src/pay_api/utils/enums.py                         31      0      0      0   100%
src/pay_api/utils/errors.py                        24      0      0      0   100%
src/pay_api/utils/handlers.py                       7      0      0      0   100%
src/pay_api/utils/logging.py                        9      0      2      0   100%
src/pay_api/utils/run_version.py                   10      0      2      0   100%
src/pay_api/utils/trace.py                          5      0      0      0   100%
src/pay_api/utils/util.py                           8      0      0      0   100%
src/pay_api/version.py                              2      0      0      0   100%
---------------------------------------------------------------------------------
TOTAL                                            2655     11    302     10    99%
Coverage XML written to file coverage.xml
```

File: pay-api/tests/unit/api/test_payment.py [hunk 1]
```diff
@@ -309,7 +309,7 @@ def test_payment_delete_when_paybc_is_down(session, client, jwt, app):
 
     with patch('pay_api.services.oauth_service.requests.post', side_effect=ConnectionError('mocked error')):
         rv = client.delete(f'/api/v1/payment-requests/{pay_id}', headers=headers)
-        assert rv.status_code == 400
+        assert rv.status_code == 202
```
Comment (sutherlanda) [inline pay-api/tests/unit/api/test_payment.py]: I just learned now what a 202 status code means :)

File: jobs/update-stale-payment/requirements/prod.txt [hunk 2]
```diff
@@ -15,5 +15,6 @@ jsonschema
 requests
 
 -e git://github.com/pwei1018/jaeger-client-python.git@186f14e14758273ed108508c0d388a4f4de5c75b#egg=jaeger-client
--e git+https://github.com/pwei1018/sbc-common-components.git#egg=sbc-common-components-1.0.0&subdirectory=python
-git+https://github.com/bcgov/sbc-pay.git#egg=pay-api&subdirectory=pay-api
+-e git+https://github.com/bcgov/sbc-common-components.git#egg=sbc-common-components-1.0.0&subdirectory=python
+# -e git+https://github.com/bcgov/sbc-pay.git#egg=pay-api&subdirectory=pay-api
+-e git+https://github.com/pksumesh/sbc-pay.git@delete_payment#egg=pay-api&subdirectory=pay-api
```
Comment (saravankumarpa) [inline jobs/update-stale-payment/requirements/prod.txt]: the url looks like referring to a fork? is that okay?

Comment (pksumesh) [reply jobs/update-stale-payment/requirements/prod.txt]: That will be changed in next PR, made that change to deploy in DEV as development won't have the latest changes to use in the job.

**Pull request merged**